src/Serialization/SerializationWriterToStringTrait.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+namespace Microsoft\Kiota\Abstractions\Serialization;
+
+use DateInterval;
+use DateTime;
+use DateTimeInterface;
+
+/**
+ * This Trait contains common to-string conversion for common types.
+ */
+trait SerializationWriterToStringTrait
+{
+    public function getDateIntervalValueAsString(DateInterval $value): string
```

Comment:
shouldn't this method be directly in the class instead?

Reply:
Do you mean in the `DateInterval` class?

Reply:
yes

Reply:
The class only provides a `format` method that allows you to pass the format string to convert to string. The class doesn't implement a `__toString` method.

Reply:
could we simply implement to string and use that in the serialization? would that cause other side effects?

Reply:
That would mean creating a custom date interval class since we can implement interfaces for classes we don't control. That might technically mean a breaking change.